pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -536,6 +553,15 @@ func (r *ReconcileBareMetalHost) actionInspecting(prov provisioner.Provisioner,
 		return recordActionFailure(info, metal3v1alpha1.InspectionError, provResult.ErrorMessage)
 	}
 
+	// Delete inspect annotation if exists
+	if hasInspectAnnotation(info.host) {
+		clearInspectAnnotation(info.host)
+		if err := r.client.Update(context.TODO(), info.host); err != nil {
+			return actionError{errors.Wrap(err, "failed to remove inspect annotations from host")}
+		}
```

Comment:
Why do we need to write this immediately? It seems like it could wait until the final write at the end of the reconciliation just like ~everything else?

Reply:
Fixed.